PocketCastsTests/Tests/Utilities/PaidFeatureTests.swift

```diff
@@ -0,0 +1,80 @@
+import XCTest
+
+@testable import podcasts
+import PocketCastsServer
+
+final class PaidFeatureTests: XCTestCase {
+    override func setUp() {
+        TestSubscriptionHelper.noSubscription()
+    }
+
+    func testFreeFeatureIsAlwaysUnlocked() {
+        assertTierValues(tier: .none, free: true, plus: true, patron: true)
+    }
+
+    func testPlusFeatureIsUnlockedForPlusAndPatron() {
+        assertTierValues(tier: .plus, free: false, plus: true, patron: true)
+    }
+
+    func testPatronFeatureIsUnlockedForOnlyPatron() {
+        assertTierValues(tier: .patron, free: false, plus: false, patron: true)
+    }
+
+    func assertTierValues(tier: SubscriptionTier, free: Bool, plus: Bool, patron: Bool) {
+        let feature = feature(tier: tier)
+
+        XCTAssertEqual(feature.isUnlocked, free)
+
+        TestSubscriptionHelper.activePlus()
+        XCTAssertEqual(feature.isUnlocked, plus)
+
+        TestSubscriptionHelper.activePatron()
+        XCTAssertEqual(feature.isUnlocked, patron)
+    }
+
+    private func feature(tier: SubscriptionTier) -> PaidFeature {
+        PaidFeature(tier: tier, subscriptionHelper: TestSubscriptionHelper.self)
+    }
+}
+
+private class TestSubscriptionHelper: SubscriptionHelper {
+    static var _hasActiveSubscription: Bool = false
+    static var _subscriptionType: SubscriptionType = .none
+    static var _subscriptionTier: SubscriptionTier = .none
+
+    override class func hasActiveSubscription() -> Bool {
+        _hasActiveSubscription
+    }
+
+    override class func subscriptionType() -> SubscriptionType {
+        _subscriptionType
+    }
+
+    override class var subscriptionTier: SubscriptionTier {
+        set {
+            _subscriptionTier = newValue
+        }
+
+        get {
+            _subscriptionTier
+        }
+    }
+
+    static func noSubscription() {
+        _hasActiveSubscription = false
+        _subscriptionTier = .none
+        _subscriptionType = .none
+    }
+
+    static func activePlus() {
+        _hasActiveSubscription = true
+        _subscriptionTier = .plus
+        _subscriptionType = .plus
+    }
+
+    static func activePatron() {
+        _hasActiveSubscription = true
+        _subscriptionTier = .patron
+        _subscriptionType = .plus
```

Comment:
should this be `.supporter`?

Reply:
Nah, supporter is from an old feature and isn't used anymore.

Reply:
This is a great question, and reminds me that I can actually remove some code here because its not needed. 

The short answer is, no because the `.supporter` value is for the paid podcasts feature that isn't being used. It's kept in the enum just to maintain compatibility. 

The longer answer of why I was setting the 3 propergties is because the subscriptionType is the old way of checking for plus, and there is [fallback handling](https://github.com/Automattic/pocket-casts-ios/blob/4528d2fc405a8b49c06659c85ea3e2937e9fcc28/Modules/Server/Sources/PocketCastsServer/Public/SubscriptionHelper.swift#L17-L22) in the activeTier code that allowed us to use the new `SubscriptionTier` even when it wasn't on the production yet servers. 

And since the SubscriptionType only ever returns `none` or `plus`, even if the user has Patron (Internal Ref: `pdeCcb-2fL#comment-2074`), using `.plus` for this value is technically correct and reflects what we receive from the server for patron (`tier = patron, type = plus`)

However for the purposes of testing there isn't a need to set the subscriptionType so I've removed it here: 39e7b25274da6e7a982d56ceac3ce4230da5a5e1

I believe the tier changes are now on production so I can look into removing the fallback code in `activeTier` since it shouldn't be needed anymore.